web-animations-2/Overview.bs

```diff
@@ -2006,6 +2045,34 @@ partial interface AnimationTimeline {
 
 </div>
 
+<h3 id="the-animation-interface">The <code>Animation</code> interface</h3>
+<pre class="idl">
+[Exposed=Window]
+partial interface Animation : EventTarget {
+    attribute CSSNumberish?       startTime;
+    attribute CSSNumberish?       currentTime;
+};
+</pre>
+
+<div class='attributes'>
+:   <dfn attribute for=Animation>startTime</dfn>
+::  The [=start time=] of this animation. When associated with a
+    [=progress-based timeline=], [=start time=] must be returned as a
+    <a href="https://drafts.css-houdini.org/css-typed-om/#cssnumericvalue">
+    CSSNumericValue</a> with percent units. Otherwise [=start time=] must be
+    returned as a double value, representing the time in units of milliseconds.
```

Comment:
It feels a little out of place to declare in the interface the types that must be returned. I would have expected these types to be the implicit result of starting an animation on or switching an animation to a progress based timeline.

Reply:
CSSNumerish is naturally ambiguous since it could be a double or a CSSNumericValue, and when expressed as a CSSNumericValue it could have any compatible units (e.g. 's' vs 'ms').  Internally, it doesn't matter how we should express time, but should have a canonical form for API return values. Perhaps this belong in a normalization procedure that is called at the end of "getting the current/start time".

Reply:
Perhaps this should be clarified in an addendum to https://www.w3.org/TR/web-animations-1/#time-value-section.

Reply:
Yeah, this is why Typed OM never uses CSSNumberish for method return types, and when it uses it for an attribute, it's very explicit about what is returned by the getter.

Reply:
Ack, sounds like it is appropriate to be explicit here about the type that should be expected.